examples_utils/benchmarks/command_utils.py

```diff
@@ -0,0 +1,186 @@
+    variations are instead created by every possible combination of parameters.
+    For example:
+        - batch_size: {3, 4}
+        - gradient_accumulation: {150, 120}
+    Will result in 4 variants:
+        - variant 1: batch_size=3, gradient_accumulation=150
+        - variant 2: batch_size=3, gradient_accumulation=120
+        - variant 3: batch_size=4, gradient_accumulation=150
+        - variant 4: batch_size=4, gradient_accumulation=120
+    Dicts can have varying numbers of values, the above rule will still apply.
+
+    Parameters
+    ----------
+    benchmark_dict:
+        benchmark entry itself in yaml file
+
+    Returns
+    -------
+    variants: list
+        List of all variants from the given benchmark entry
+    """
+
+    variants = [{}]
+    if "parameters" in benchmark_dict:
+        # Matching parameters as listed (see notes)
+        if isinstance(benchmark_dict["parameters"], list):
+            names = benchmark_dict["parameters"][0]
+
+            variants = []
+            for vals in benchmark_dict["parameters"][1:]:
+                new_variant = {}
+                for k, v in zip(names, vals):
+                    new_variant[k] = str(v)
+
+                variants.append(new_variant)
+
+        # Matching parameters by every possible combination
+        elif isinstance(benchmark_dict["parameters"], dict):
+            for name, valstr in benchmark_dict["parameters"].items():
+                current_variants = variants
+                variants = []
+
+                vals = str(valstr).split(",")
+                for v in vals:
+                    if v is not None:
+                        for c in current_variants:
+                            x = c.copy()
+                            x[name] = v
+                            variants.append(x)
+
+        else:
+            raise ValueError("'parameters' defined are neither a list nor a dict")
+
+    return variants
+
+
+def get_benchmark_variants(benchmark_name: str, benchmark_dict: dict) -> list:
+    """
+    Get all named variations of a benchmark.
+
+    Parameters
+    ----------
+    benchmark_name: str
+        Benchmarks name as given in the spec yaml file
+
+    benchmark_dict:
+        benchmark entry itself in yaml file
+
+    Returns
+    -------
+    variations:
+        List of all possible variants from this benchmark
+    :returns [String] of all possible variants
+    """
+
+    # Create variants from benchmark
+    variant_names = create_variants(benchmark_dict)
+
+    variants = []
+    for variant in variant_names:
+        work_str = benchmark_name
+
+        for k in sorted(variant.keys()):
+            work_str += "_" + k + "_" + variant[k]
+
+        variants.append({"name": work_str, "config": variant})
+
+    return variants
+
+
+def formulate_benchmark_command(
+        benchmark_dict: dict,
+        variant_dict: dict,
+        ignore_wandb: bool,
+        compile_only: bool,
+        logger: logging.Logger,
```

Comment:
Usually you don't need to pass a logger, you just import it, use it and it should work.

Reply:
Its globally available by default??? wow. 

Will treat it as such and remove from function parameters/signatures

Reply:
if I have multiple files, how would this work? Do I need to instantiate a new logger in each py file for the functions inside to recognise it?